Logback 929

eyalfa opened the pull request:
attempt to fix http://jira.qos.ch/browse/LOGBACK-929
basically daemonize threads created by logback's execution context.

tony19 commented:
Thanks for the contribution, @eyalfa. Please add a unit test, squash/rebase your commit on top of master so that this PR contains only your commits, and add a comment here with release notes. Please refer to our [contributing guidelines](https://github.com/qos-ch/logback/blob/master/CONTRIBUTING.md) for more info.

eyalfa commented:
Hi Tony, what would you like to see in a unit test?
What I have in mind currently is this: explicitly obtain a reference to the threadpool and execute a callable that simply checks that the current thread is a daemon thread.
Is this enough? I can't see a way to test that the JVM is shutting properly, can you think of a way?

eyalfa commented:
Tony, I've added a unit-test and fixed the styling issue you commented about.
can you please have a look?

I'm still trying to figure out how to squash my branch into a single commit, will do that once I have your approval regarding the contents.

BTW, test ch.qos.logback.core.rolling.helper.RollingCalendarTest#testVaryingNumberOfDailyPeriods is locale dependent, it fails when executing on a timezone other thes CEST (actually specified in the code), this seems like a bad practice, do you agree?

tony19 commented:
The `isDaemon` check is probably sufficient for now. I'll have to look into a standard way of running integration tests that are executed after unit tests. These tests would run a sample logback app, setup a SocketAppender/FileAppender, attempt to exit, verify the application exits successfully within some timeframe (2 secs?), and then verify that the socket/file is closed.

tony19 commented:
There are several ways to rebase your changes. Your PR really only contains 2 relevant commits (the original change plus a typo fix). Rebase your PR branch (using the `git rebase` command) so that your branch contains only those two commits.

The CEST issue with `RollingCalendarTest` should be fixed by 19cde4d58e8264ab2086730d8e7e3cd8757007ca.

eyalfa commented:
ok, changed the spacing per your request.
I'm using a list as a simple place-holder, seemed simpler than defining a single-property bean (Java don't have c++ like references).

I'll squash everything into a single commit now, hopefully all is well and ready to be merged... :-)

BTW, commit 19cde4d is merged into my branch and the RollingCalendarTest fails unless I'm setting user.timezone to CEST, as far as I can tell I'm running under Zone 'Asia/Jerusalem', might worth a check on your side.

eyalfa commented:
squashed

eyalfa commented:
Tony, can you please explain? :-) what do you mean by spacing and how can this be fixed (I'm using intellij idea)

tony19 commented:
I've clarified my comments. Your best bet is to match the code around you. You must be using custom style settings in IDEA because mine doesn't do that out of the box.

eyalfa commented:
spaced

eyalfa commented:
squashed again.
Tony, any idea what caused the build failure? the diff between my previous (successful) build and the last one (after rebase+squash) is empty.

eyalfa commented:
this is strange, just amended the commit message.
seems like a non deterministic build issue

tony19 commented:
@eyalfa, please provide a comment here with release notes (in HTML). Thanks.

eyalfa commented:
is this ok?
<p>All threads opened by <code>ch.qos.logback.core.util.ExecutorServiceUtil#THREAD_FACTORY</code> are now daemons.
This fixes <a href="http://jira.qos.ch/browse/LOGBACK-929">LOGBACK-929</a>.</p>
